### PR TITLE
Adding basic automated test infrastructure

### PR DESCRIPTION
**Pull request by ghop02**
- Updates requirements files + adds test requirements file
- Adds a SETUP.md file for instructions on setting up environment
- Adds a makefile with some basic commands.  Right now I added super loose linting, no style checks at all right now.
- Adds a github action that runs tests

### CODE REVIEW

File: .github/workflows/python_unit_tests.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
```
Comment (playermanny2) [inline .github/workflows/python_unit_tests.yml]: LGTM --

Question: This seems like a duplicate dependency installation as the make file also handles requirements install?

Also where is matrix.python-version env coming from?

Comment (ghop02) [reply .github/workflows/python_unit_tests.yml]: and It is duplicate in the sense of the code is duplicated. I can just use a make command to simplify!

Probably don't need the matrix here, but theoretically we could run on multiple python environments
https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix

Comment (playermanny2) [reply .github/workflows/python_unit_tests.yml]: Ahh i see. I'm good with this, thanks for handling this!

#7

File: requirements_test.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Requirements file containing packages needed for testing.
+# These should *not* be installed in non-test environments.
```
Comment (mikelehen) [inline requirements_test.txt]: FWIW- I think everybody running the model should also be running tests, so I think separating out test vs. non-test requirements is probably overkill, though not particularly harmful either.  So I'm fine if you want to keep them separated.

File: .github/workflows/python_unit_tests.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Python Unit Tests
```
Comment (mikelehen) [inline .github/workflows/python_unit_tests.yml]: Just to sanity-check, have you been able to test this somehow?  I'm not very familiar with GitHub workflows, so not sure how you go about vetting them before committing.

File: requirements_test.txt (hunk 2 above)
Comment (ghop02) [reply requirements_test.txt]: Oh definitely agree around everyone running tests.  

The non-test requirements is helpful when you start deploying to production environments.  Sometimes you could have packages that you may want locally but not on production.

**Comment by crwilcox** [comment]
This LGTM in general. If folks are open to it I can make a PR to move from make to nox or tox. Mostly to help keep things to virtualenvs on folks machines. But this is a big step forward!

File: requirements_test.txt (hunk 2 above)
Comment (mikelehen) [reply requirements_test.txt]: Yeah.  There's no notion of "deploying to production" at this point.  But it doesn't hurt to keep this in place.

File: .github/workflows/python_unit_tests.yml (hunk 3 above)
Comment (ghop02) [reply .github/workflows/python_unit_tests.yml]: yep - it actually runs as soon as you add a file - so you can see in the results below

**Comment by ghop02** [comment]
Yeah I'm open to either! I haven't used tox too much, so will defer to you @crwilcox !

**Pull request merged**